core/src/main/java/org/mule/api/registry/ServiceRegistry.java

```diff
@@ -31,11 +27,9 @@
      * @param providerClass a <code>Class</code>object indicating the
      *                      class or interface of the service providers being detected.
      * @param loader        the class loader to be used to load
-     *                      provider-configuration files and instantiate provider classes,
-     *                      or <code>null</code> if the system class loader (or, failing that
-     *                      the bootstrap class loader) is to be used.
-     * @return A {@link Collection} that yields provider objects
-     * for the given service, in some arbitrary order.
+     *                      provider/configuration files and instantiate provider instances.
+     *                      If {@code null}, it will be up to the implementation to choose a {@link ClassLoader}
+     * @return A {@link Collection} that yields provider objects for the given service, in some arbitrary order.
```

Comment:
Could not be a problem if depending on the environment, providers are returned in different order?